### ISSUE TITLE
`weave run` without CIDR and IPAM disabled leaves container running unexpectedly

### ISSUE THREAD

**Issue by awh** [weaveworks/weave#889]
The error message to stderr and lack of container ID to stdout leads one to believe that their container has not started when it has (and is presumably detached from the weave network):
````
vagrant@vagrant-ubuntu-utopic-64:~/weave$ ./weave launch
c18a443dfe0f290be2e95fa7b1836ea6a25975c90d609ae4d9c66cc76b1de83c
vagrant@vagrant-ubuntu-utopic-64:~/weave$ ./weave run --name=foo -ti ubuntu
No IP address supplied (use the -iprange option on 'weave launch' to enable IP address allocation)
vagrant@vagrant-ubuntu-utopic-64:~/weave$ docker ps
CONTAINER ID        IMAGE                     COMMAND                CREATED             STATUS              PORTS                                            NAMES
d71c3f28a194        ubuntu:latest             "/bin/bash"            6 seconds ago       Up 5 seconds                                                         foo                 
c18a443dfe0f        weaveworks/weave:latest   "/home/weave/weaver    16 seconds ago      Up 16 seconds       0.0.0.0:6783->6783/tcp, 0.0.0.0:6783->6783/udp   weave               
````

**Comment by rade** [comment]
Somewhat related to #192, which only just fixed. *sigh*

**Comment by rade** [comment]
This is going to be hard to fix, since `weave run` is implemented as `docker run` followed by `weave attach`, and we do not want to kill containers on `weave attach` failure. Also, more generally, killing any application container is questionable. We have no idea what it may have done up to that point, even if we've only just started it.

**Comment by bboreham** [comment]
This particular failure should go away when #743 is done, but address allocation could still fail for other reasons.

**Comment by rade** [comment]
I should note that the proxy does not suffer from this. Because of the entrypoint rewriting, it can safely assume that the container won't have done any work until network configuration has been completed, and hence kills it if anything fails before that point.

So perhaps we shouldn't worry about this too much. Just use the proxy.

**Comment by rade** [comment]
Closed by #932.
